src/v/raft/consensus.cc

```diff
@@ -186,17 +186,22 @@ consensus::success_reply consensus::update_follower_index(
           "Append entries response send to wrong group: {}, current group: {}",
           reply.group));
     }
-    if (seq < idx.last_received_seq) {
+    if (
+      seq < idx.last_received_seq
+      && reply.last_dirty_log_index < _log.offsets().dirty_offset) {
         vlog(
           _ctxlog.trace,
-          "ignorring reordered reply from node {} - last: {} current: {} ",
+          "ignorring reordered reply {} from node {} - last: {} current: {} ",
+          reply,
           reply.node_id,
           idx.last_received_seq,
           seq);
         return success_reply::no;
     }
     // only update for in order sequences
-    idx.last_received_seq = seq;
+    if (seq >= idx.last_received_seq) {
```

Comment:
i think that this could use a comment in the commit message. before this change seq was always >= last received seq, but not that isn't true and the _name_ `last_received_seq` seems to not reflect reality any more since we might not update it with the last received seq?